### ISSUE TITLE
Hello, can this be used in English? textbsr doesn't work very well for English.

### ISSUE THREAD

**Issue by lhzwangwei** [csxmli2016/MARCONet#17]

**Comment by csxmli2016** [comment]
> _No description provided._

Yes. The released model is mainly trained using Chinese. If the application scenario is English, it is better to fine-tune it in English and Number only. Chinese contains several thousand characters, which is more difficult than English.

**Comment by wojiaoyanmin** [comment]
Thanks, But what if both Chinese and English are used in the application scenario?

**Comment by csxmli2016** [comment]
> Thanks, But what if both Chinese and English are used in the application scenario?

It is better to fine-tune the model by combining English characters and Chinese but with a relatively higher possibility for English characters.

**Comment by wojiaoyanmin** [comment]
The max character length of the model is 16, I think that's not enough for English characters,  simplely change the max charater length may not work well, is there any solutions?

**Comment by csxmli2016** [comment]
> The max character length of the model is 16, I think that's not enough for English characters, simplely change the max charater length may not work well, is there any solutions?

The max character length of 16 is mainly designed for Chinese characters, which cover larger spaces than English letters. So if you want to use it in the English characters, it is better to fine-tune on longer character length.

**Comment by wojiaoyanmin** [comment]
> The max character length of 16 is mainly designed for Chinese characters, which cover larger spaces than English letters. So if you want to use it in the English characters, it is better to fine-tune on longer character length.

Thankyou! Have you ever tried this pipeline? Did it work well?

**Comment by csxmli2016** [comment]
> > The max character length of 16 is mainly designed for Chinese characters, which cover larger spaces than English letters. So if you want to use it in the English characters, it is better to fine-tune on longer character length.
> 
> Thankyou! Have you ever tried this pipeline? Did it work well?

You can see that only the character location and classification are related to the character length. 
As for the SR process, we embed the generative structure prior into each LR character. This process has no relationship with the character length. 
So I think it would work well.

**Comment by wojiaoyanmin** [comment]
> > > The max character length of 16 is mainly designed for Chinese characters, which cover larger spaces than English letters. So if you want to use it in the English characters, it is better to fine-tune on longer character length.
> > 
> > 
> > Thankyou! Have you ever tried this pipeline? Did it work well?
> 
> You can see that only the character location and classification are related to the character length. As for the SR process, we embed the generative structure prior into each LR character. This process has no relationship with the character length. So I think it would work well.

Thank you very much!!

**Comment by wojiaoyanmin** [comment]
Hi.can you provide some suggestions about how to set these parameters when finetune the model with more English characters?

**Comment by csxmli2016** [comment]
> Hi.can you provide some suggestions about how to set these parameters when finetune the model with more English characters?

Hi, Just replace the corpus with English Characters. See the function of get_text() [here](https://github.com/csxmli2016/MARCONet/blob/58582fe5801b4ff3b5cdaf6e96c8feb3a426b68a/Train/tspgan/data/text_degradation_dataset.py#L292). It would be really simple than Chinese Character.

**Comment by wojiaoyanmin** [comment]
sorry,I forget to upload the picture, I mean the parameters below. My main goal in fine-tuning the model was to make it perform well in the Chinese scene at the same time. 
![image](https://github.com/csxmli2016/MARCONet/assets/57952327/8d21552f-9812-4801-8720-c04926c2fb4f)

**Comment by csxmli2016** [comment]
> sorry,I forget to upload the picture, I mean the parameters below. My main goal in fine-tuning the model was to make it perform well in the Chinese scene at the same time. ![image](https://private-user-images.githubusercontent.com/57952327/318163528-8d21552f-9812-4801-8720-c04926c2fb4f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE3NzAwNTAsIm5iZiI6MTcxMTc2OTc1MCwicGF0aCI6Ii81Nzk1MjMyNy8zMTgxNjM1MjgtOGQyMTU1MmYtOTgxMi00ODAxLTg3MjAtYzA0OTI2YzJmYjRmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzMwVDAzMzU1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJlM2M2YzJiNjc2YzE0ZTQ1MDIxZmZjODc1ZTJmZGUxMGExYjg4NDExMzFhZDU2OWJkODJiZmRlNDJmOTJhOWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pL-VxBS1QOAs1dfv6M090UD-c5yIOdzmrVJpwqqw584)

The parameter looks fine. The SR branch has already performed well. You may pay more attention to the ocr and encoder branckh, using a slightly larger learning rate.

**Comment by wojiaoyanmin** [comment]
THx



发自我的iPhone


------------------ Original ------------------
From: csxmli2016 ***@***.***&gt;
Date: Sat,Mar 30,2024 11:38 AM
To: csxmli2016/MARCONet ***@***.***&gt;
Cc: justababy ***@***.***&gt;, Comment ***@***.***&gt;
Subject: Re: [csxmli2016/MARCONet] Hello, can this be used in English? textbsr doesn't work very well for English. (Issue #17)





  
sorry,I forget to upload the picture, I mean the parameters below. My main goal in fine-tuning the model was to make it perform well in the Chinese scene at the same time. 
  
The parameter looks fine. The SR branch has already performed well. You may pay more attention to the ocr and encoder branckh, using a slightly larger learning rate.
